idb/postgres/internal/writer/writer.go

```diff
@@ -278,17 +313,33 @@ func getTransactionParticipants(stxnad transactions.SignedTxnWithAD) []basics.Ad
 	return res
 }
 
-func addTransactionParticipation(block *bookkeeping.Block, batch *pgx.Batch) error {
-	for i, stxnad := range block.Payset {
-		// TODO: replace with a function from go-algorand.
-		participants := getTransactionParticipants(stxnad.SignedTxnWithAD)
+func addInnerTransactionParticipation(stxnad transactions.SignedTxnWithAD, round, intra uint64, batch *pgx.Batch) uint64 {
+	finalIntra := intra
+	for _, itxn := range stxnad.ApplyData.EvalDelta.InnerTxns {
+		// Only search inner transactions by direct participation.
+		// TODO: Should inner app calls be surfaced by their participants?
+		participants := getTransactionParticipants(itxn, false)
 
 		for j := range participants {
-			batch.Queue(addTxnParticipantStmtName, participants[j][:], uint64(block.Round()), i)
+			batch.Queue(addTxnParticipantStmtName, participants[j][:], round, finalIntra)
 		}
+
+		finalIntra = addInnerTransactionParticipation(itxn, round, finalIntra+1, batch)
```

Comment:
Add 1? It would probably be better to have `addInnerTransactionParticipation()` return the index following the last.

Reply:
I prefer this way, I think it's a better pattern for recursive functions

Reply:
`addInnerTransactionParticipation` does return the index following the last, sorry for the confusion.